File: app/src/main/java/com/slq/tokfm/PodcastAdapter.java
package com.slq.tokfm;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.slq.tokfm.model.Podcast;

import java.util.List;

public class PodcastAdapter extends BaseAdapter {

    private final Context context;
    private final List<Podcast> items;

    public PodcastAdapter(Context context, List<Podcast> items) {
        this.context = context;
        this.items = items;
    }

    @Override
    public int getCount() {
        return items.size();
    }

    @Override
    public Object getItem(int position) {
        return items.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        View view = convertView;

        if (view == null) {
            view = LayoutInflater.from(context).inflate(R.layout.podcast_row, null);
        }

        Podcast podcast = (Podcast) getItem(position);

        TextView podcastName = view.findViewById(R.id.podcast_name);
        TextView seriesName = view.findViewById(R.id.series_name);
        TextView podcastDescription = view.findViewById(R.id.podcast_description);
        TextView published = view.findViewById(R.id.published);
        TextView size = view.findViewById(R.id.size);
        ImageView image = view.findViewById(R.id.podcast_image);

        podcastName.setText(podcast.getTitle());
        seriesName.setText(podcast.getSeries());
        podcastDescription.setText(podcast.getLeader());
        published.setText(podcast.getDateTime());
        size.setText(podcast.getLength());
//        image.setImageBitmap(podcast.getImage());

        return view;
    }
}

File: app/src/main/java/com/slq/tokfm/PodcastAdapter.kt
package com.slq.tokfm

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import android.widget.TextView


class PodcastAdapter(private val context: Context, private val items: List<Podcast>) : BaseAdapter() {

    override fun getCount() = items.size

    override fun getItem(position: Int): Podcast = items[position]

    override fun getItemId(position: Int) = position.toLong()

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        // inflate the layout for each list row
        val view: View = convertView ?: LayoutInflater.from(context).inflate(R.layout.podcast_row, parent, false)
        return getView(position, view)
    }

    fun getView(position: Int, view: View): View {
        // get current item to be displayed
        val currentItem = getItem(position)

        // get the TextView for item name and item description
        val podcastName = view.findViewById(R.id.podcast_name) as TextView
        val seriesName = view.findViewById(R.id.series_name) as TextView
        val podcastDescription = view.findViewById(R.id.podcast_description) as TextView
        val published = view.findViewById(R.id.published) as TextView
        val size = view.findViewById(R.id.size) as TextView
//        val podcastSeriesDescription = view.findViewById(R.id.podcast_series_description) as TextView
        val podcastImage = view.findViewById(R.id.podcast_image) as ImageView


        //sets the text for item name and item description from the current item object
        seriesName.text = currentItem.series_name
        podcastName.text = currentItem.podcast_name
        if(currentItem.podcast_description.isNotEmpty()) {
            podcastDescription.text = currentItem.podcast_description
        } else {
            podcastDescription.visibility = View.GONE
        }
        published.text = currentItem.podcast_emission_text
        size.text = currentItem.podcast_size_mb
//        podcastSeriesDescription.text = currentItem.series_description
        podcastImage.setImageBitmap(currentItem.podcast_image)

        // returns the view for the current row
        return view
    }

}